src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/unit/reporting/generator/BedUtilisationReportGeneratorTest.kt

```diff
@@ -877,7 +879,678 @@ class BedUtilisationReportGeneratorTest {
     // 4 days for relevantVoidStraddlingStartOfMonth
     // 6 days for relevantVoidStraddlingEndOfMonth
     assertThat(result[0][BedUtilisationReportRow::totalBookedDays]).isEqualTo(7)
-    assertThat(result[0][BedUtilisationReportRow::totalDaysInTheMonth]).isEqualTo(30)
+    assertThat(result[0][BedUtilisationReportRow::bedspaceOnlineDays]).isEqualTo(30)
     assertThat(result[0][BedUtilisationReportRow::occupancyRate]).isEqualTo(7.toDouble() / 30)
   }
+
+  @Test
+  fun `bedspaceStartDate show bedspace created date`() {
+    val apArea = ApAreaEntityFactory()
+      .produce()
+
+    val probationRegion1 = ProbationRegionEntityFactory()
+      .withApArea(apArea)
+      .produce()
+
+    val localAuthorityArea1 = LocalAuthorityEntityFactory().produce()
+
+    val premises = TemporaryAccommodationPremisesEntityFactory()
+      .withLocalAuthorityArea(localAuthorityArea1)
+      .withProbationRegion(probationRegion1)
+      .produce()
+
+    val room = RoomEntityFactory()
+      .withPremises(premises)
+      .produce()
+
+    val bed = BedEntityFactory()
+      .withRoom(room)
+      .withCreatedAt { OffsetDateTime.parse("2024-01-16T14:03:00+00:00") }
+      .produce()
+
+    every {
+      mockBookingRepository.findAllByOverlappingDateForBed(
+        LocalDate.parse("2024-02-01"),
+        LocalDate.parse("2024-02-29"),
+        bed,
+      )
+    } returns emptyList()
+    every {
+      mockLostBedsRepository.findAllByOverlappingDateForBed(
+        LocalDate.parse("2024-02-01"),
+        LocalDate.parse("2024-02-29"),
+        bed,
+      )
+    } returns emptyList()
+
+    val result = bedUtilisationReportGenerator.createReport(
+      listOf(bed),
+      BedUtilisationReportProperties(ServiceName.temporaryAccommodation, null, 2024, 2),
+    )
+
+    assertThat(result.count()).isEqualTo(1)
+    assertThat(result[0][BedUtilisationReportRow::bedspaceStartDate]).isEqualTo(LocalDate.parse("2024-01-16"))
+  }
+
+  @Test
+  fun `bedspaceEndDate show bedspace end date when it is not null`() {
+    val apArea = ApAreaEntityFactory()
+      .produce()
+
+    val probationRegion1 = ProbationRegionEntityFactory()
+      .withApArea(apArea)
```

Comment:
If possible, Data setup for bed can be refactored into reusable method, so that we can avoid duplicate code.

Reply:
Its not a big win as the premises will be needed later